### ISSUE TITLE
ListView bound to ISupportIncrementalLoading collection causes application crash.

### ISSUE THREAD

**Issue by chris-atk** [microsoft/microsoft-ui-xaml#6883]
### Describe the bug

I'm getting a crash to desktop and the following error message when binding ListView or GridView to an item source that implements ISupportIncrementalLoading.

```
Exception thrown: 'System.ArgumentNullException' in System.Linq.Expressions.dll
Exception thrown at 0x76FAFD62 (KernelBase.dll) in WinUi3Blank.exe: WinRT originate error - 0x80004003 : 'Value cannot be null. (Parameter 'constructor')'.
```

I have reproduced this issue in a minimal project using the VS2022 built in WinUI 3 project template. The behaviour is the same when targeting x86 and x64 and when targeting Windows 10 19044 and 18363 builds.

I have also tried with all Nuget packages updated to the latest versions, I get the same behaviour.

It seems like a viable workaround is to switch from incremental loading to random loading, the `IItemsRangeInfo` seems to work fine.

### Steps to reproduce the bug

1. Create a new VS2022 project using the `Blank App, Packaged (WinUI 3 in Desktop)` template
2. Create a new class that implements `ISupportIncrementalLoading` based on the [example from the documentation](https://docs.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.data.isupportincrementalloading?view=winui-3.0#examples)
3. Create a read only property on MainWindow and initialise it with an instance of your `ISupportIncrementalLoading` class.
4. Add a ListView to the main windows xaml and bind ItemsSource to the property.
5. Run the app

### Expected behavior

The app should show the items in the item source but I get a crash to desktop.

### NuGet package version

WinUI 3 - Windows App SDK 1.0

### Windows app type

- [ ] UWP
- [X] Win32

### Device form factor

Desktop

### Windows version

Windows 10 (21H2): Build 19044, Windows 10 (1909): Build 18363

**Comment by LasmGratel** [comment]
Facing same issue here, I'm using IncrementalLoadingCollection from CommunityToolkit.WinUI and it throws the exact same exceptions.

**Comment by dylech30th** [comment]
Me and one of my friends are also facing this problem, I suspect this is relevant to the OS version, since the code and project configuration didn't change at all after the last time I build it, and the last build works perfectly fine, but once I rebuild it, this problem occurred, the only thing changed during these two builds is the OS Version.

**Comment by dylech30th** [comment]
This is a severe issue, the main functionality of those content-heavy apps are almost broken at all, this should not happen in a so-called production-ready framework/OS, please address it quickly

**Comment by ojhad** [comment]
@MikeHillberg is this a known issue?

**Comment by LasmGratel** [comment]
Any updates on this?

**Comment by Pseudocode99** [comment]
Any updates on this?

**Comment by huoyaoyuan** [comment]
It crashes whenever `HasMoreItems` returns true.

New compiled code with 1.0.0/1.0.1/1.1.0-preview all crashes, in both Debug and Release config. However, previously compiled package does not crash. Reverting to corresponding commit still crashes. The only differences should be .NET SDK patch version and OS patch version.

Current Windows version: 19043.1586

**Comment by huoyaoyuan** [comment]
Confirmed it's a CsWinRT issue, like #4414.

Workaround: add following into your project file in ItemGroup:
```csproj
<FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" TargetingPackVersion="10.0.19041.22" RuntimeFrameworkVersion="10.0.19041.22" />
<!-- Change 19041 to your target SDK version -->
```

When adding to directory settings, remember to add into `Directory.Build.targets`, not `.props`.

**Comment by manodasanW** [comment]
Per my response in https://github.com/microsoft/CsWinRT/issues/1165, this regression has been fixed in the **.24** version of the Windows SDK projection.  It is included in the upcoming .NET SDK servicing update.  But in the meantime, you can set the below property to pick up the .24 version of the Windows SDK projection and unblock yourself.  Once the .NET SDK update comes out and you move to it, you should remove the property as it fixes it to that version.

Let me know if it resolves your issue.

`<WindowsSdkPackageVersion>10.0.<Target_Windows_Sdk_Version>.24</WindowsSdkPackageVersion>`

i.e. if you are targeting 19041, you can set the following the following property:

`<WindowsSdkPackageVersion>10.0.19041.24</WindowsSdkPackageVersion>`

**Comment by manodasanW** [comment]
The above workaround is no longer needed.  This issue should be fixed in the .NET SDK servicing updates (6.0.202, 6.0.104, 5.0.407, 5.0.213) that came out yesterday.

**Comment by felinepunch** [comment]
This problem continues in the .NET SDK 6.0.400

**Comment by chris-atk** [comment]
> This issue is stale because it has been open 180 days with no activity. Remove stale label or comment or this will be closed in 5 days.

Bump
